### ISSUE TITLE
About custom dataset.

### ISSUE THREAD

**Issue by willpat1213** [SysCV/sam-hq#55]
There is a custom dataset, each picture contains multiple objects, how to modify the dataloader or model to train such a dataset?

**Comment by ymq2017** [comment]
Hi, for a multi-object dataset, you could simply modify the dataloader to sample one mask each time randomly. 
Or you could load all masks and use them as prompts. Then we can change the prompt input to multi-mask, multi-box, multi-points. And the multi-mask output can be used to calculate loss with input multi GT mask.
The latter requires a larger code change and would work better.

**Comment by willpat1213** [comment]
Thanks for your reply!
Another question is how do I design the loss in this case?

**Comment by Rockwangyu** [comment]
> 有一个自定义数据集，每张图片包含多个对象，如何修改 dataloader 或模型来训练这样的数据集？

hi, have you solved this problem? I also meet this problem but I do not  how to modify the dataloader to to fit my own dataset

**Comment by sym330** [comment]
hi, have you solved this problem?

**Comment by sjc6834** [comment]
Thank you very much for your concern and reminder. The problem has been solved.

---- Replied Message ----
| From | ***@***.***> |
| Date | 12/06/2024 15:52 |
| To | SysCV/sam-hq ***@***.***> |
| Cc | sjc6834 ***@***.***>,
Manual ***@***.***> |
| Subject | Re: [SysCV/sam-hq] About custom dataset. (Issue #55) |

hi, have you solved this problem?

**Comment by sym330** [comment]
> Thank you very much for your concern and reminder. The problem has been solved.
> 非常感谢您的关心和提醒。问题已经解决了。
> […](#) [...](#)
> ---- Replied Message ---- | From | ***@***.***> | | Date | 12/06/2024 15:52 | | To | SysCV/sam-hq ***@***.***> | | Cc | sjc6834 ***@***.***>, Manual ***@***.***> | | Subject | Re: [SysCV/sam-hq] About custom dataset. (Issue #55) | hi, have you solved this problem? — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

Could you kindly let me know how it was resolved? Thank you very much!
